examples/coappearances/coappearances.cpp

```diff
@@ -212,6 +212,108 @@ convert( const char* json_path, const char* archive_path )
     builder.set_strings( {strings.data( ), strings.size( )} );
 }
 
+// Note: The linear search below is slow for big graphs for obvious reasons, where it is much better
+// to build a lookup table. However, for our small graph this works fine.
+std::vector< uint32_t >
+get_neighbors( co::Graph graph, uint32_t vertex_ref )
+{
+    std::vector< uint32_t > neighbors;
+    for ( auto e : graph.edges( ) )
+    {
+        if ( e.a_ref == vertex_ref )
+        {
+            neighbors.push_back( e.b_ref );
+        }
+        else if ( e.b_ref == vertex_ref )
+        {
+            neighbors.push_back( e.a_ref );
+        }
+    }
+    return neighbors;
+}
+
+size_t
+calculate_num_connected_components( co::Graph graph )
+{
+    size_t num_connected_components = 0;
+
+    std::vector< bool > visited( graph.vertices( ).size( ) );
+    uint32_t num_visited = 0;
+    std::vector< uint32_t > stack;
+
+    while ( num_visited != graph.vertices( ).size( ) )
```

Comment:
you can just keep the inner for loop and get rid of the while loop (the one over vertex_ref), as well as num_visited:
Instead of break just do continue

Reply:
Done.